### PR TITLE
Materialize-css

### PR DESCRIPTION
**Pull request by kehtolaulu**

### CODE REVIEW

File: src/components/auth/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    }
+
+    handlePasswordChange = (e) => {
+        let newPassword = e.target.value;
+        if (newPassword.length < 6) {
+            this.setState({ password: e.target.value });
+            this.setState({ passwordError: 'Password must be 6 chars min' });
+        } else {
+            this.setState({ password: e.target.value });
```
Comment (Dimanaux) [inline src/components/auth/SignUp.js:40]: move `this.setState({ password: e.target.value });` out of if statement

File: src/components/auth/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        } else {
+            this.setState({ username: e.target.value });
+            this.setState({ usernameError: '' })
+        }
+    }
+
+    handlePasswordChange = (e) => {
+        let newPassword = e.target.value;
+        if (newPassword.length < 6) {
+            this.setState({ password: e.target.value });
```
Comment (Dimanaux) [inline src/components/auth/SignUp.js:37]: `e.target.value` can be replaced with `newPassword`

File: src/components/auth/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        };
+    }
+
+    disabled = () => {
+        return this.state.passwordError || this.state.usernameError || this.state.passwordTwiceError;
+    }
+
+    handleUsernameChange = (e) => {
+        let newUsername = e.target.value;
+        if (newUsername.length < 6) {
```
Comment (Dimanaux) [inline src/components/auth/SignUp.js:25]: you have duplicated code here and in signUp function

File: src/components/auth/SignUp.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            password: '',
+            passwordTwice: '',
+            usernameError: '',
+            passwordError: '',
+            passwordTwiceError: ''
+        };
+    }
+
+    disabled = () => {
+        return this.state.passwordError || this.state.usernameError || this.state.passwordTwiceError;
```
Comment (Dimanaux) [inline src/components/auth/SignUp.js:20]: 😍

File: src/components/auth/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            username: '',
+            password: '',
+            passwordTwice: '',
+            usernameError: '',
+            passwordError: '',
+            passwordTwiceError: ''
+        };
+    }
+
+    disabled = () => {
```
Comment (Dimanaux) [inline src/components/auth/SignUp.js:19]: Let's call it `anyErrors` or `buttonIsDisabled` or `canSignUp`. Just disabled is quite unclear

File: src/components/toDo/ToDo.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            <label>
+                <input className="done"
+                    type="checkbox"
+                    checked={status === "done"}
+                    onChange={onStatusChange}>
+                </input>
+                <span></span>
+            </label>
+        </form>
+        <span className={"todo-" + status}>
```
Comment (Dimanaux) [inline src/components/toDo/ToDo.js:15]: Very good polymorphism usage 👍

File: src/components/toDo/ToDo.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            {text}
+        </span>
+        <span>
+            <button className="delete-button btn-flat black-text" onClick={onToDoDelete}>
+                <i className="material-icons">delete</i>
+            </button>
+        </span>
+        <hr></hr>
```
Comment (Dimanaux) [inline src/components/toDo/ToDo.js:23]: `<hr />`

File: src/components/toDo/ToDoList.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import ToDo from './ToDo'
+
+class ToDoList extends React.Component {
+    render() {
+        let todoNodes = this.props.data.map(todo => (
+            <ToDo
```
Comment (Dimanaux) [inline src/components/toDo/ToDoList.js:7]: ⭐

File: src/components/toDo/ToDo.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+const ToDo = ({text, status, onStatusChange, onToDoDelete}) => (
```
Comment (Dimanaux) [inline src/components/toDo/ToDo.js:3]: 🥇

File: src/components/auth/SignUp.js (hunk 4 above)
Comment (kehtolaulu) [reply src/components/auth/SignUp.js:20]: ?

File: src/components/auth/SignUp.js [hunk 10]
```diff
@@ -60,17 +59,12 @@ class SignUp extends React.Component {
...
+            this.props.history.push("/login");
         })
-            .then(response => { // eslint-disable-line no-unused-vars
-                this.props.history.push("/login");
-            })
             .catch((error) => {
-                console.log(error);
                 this.setState({ errorText: error });
-            });
+            });;
```
[the remark this answers is not in the document]
Comment (kehtolaulu) [reply src/components/auth/SignUp.js:67]: I've got eslint

File: src/components/toDo/ToDo.js (hunk 6 above)
Comment (kehtolaulu) [reply src/components/toDo/ToDo.js:15]: ?

File: src/components/toDo/ToDo.js (hunk 7 above)
Comment (kehtolaulu) [reply src/components/toDo/ToDo.js:23]: vsc doesn't think so

File: src/components/toDo/ToDoBox.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
+import ToDoList from './ToDoList';
+import ToDoForm from './ToDoForm';
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { getToDos, saveToDo, deleteToDo, updateToDo } from '../../api/todos';
+
+class ToDoBox extends React.Component {
```
[the remark this answers is not in the document]
Comment (kehtolaulu) [reply src/components/toDo/ToDoBox.js:7]: Already thought about it, but that don't matter

File: src/components/toDo/ToDoForm.js [hunk 12]
```diff
@@ -22,24 +22,19 @@ class ToDoForm extends React.Component {
...
+                <form className="todo-form" onSubmit={this.handleSubmit}>
                     <input
+                        className="input-field"
                         size="50"
                         type="text"
                         placeholder="Write to do here!"
                         value={this.state.text}
                         onChange={this.handleTextChange}
                     />
                     <input type="submit" value="Post" style={{ display: "none" }} />
```
[the remark this answers is not in the document]
Comment (kehtolaulu) [reply src/components/toDo/ToDoForm.js:37]: How am I supposed to learn all this by heart?

File: src/components/toDo/ToDo.js (hunk 6 above)
Comment (kehtolaulu) [reply src/components/toDo/ToDo.js:15]: ?

File: src/components/toDo/ToDoBox.js [hunk 13]
```diff
@@ -68,18 +48,10 @@ class ToDoBox extends React.Component {
         } else {
```
[the remark this answers is not in the document]
Comment (kehtolaulu) [reply src/components/toDo/ToDoBox.js:48]: But not API

File: src/components/toDo/ToDoBox.js [hunk 14]
```diff
@@ -14,41 +13,22 @@ class ToDoBox extends React.Component {
...
-        fetch(apiUrl + '/todos', {
-            method: 'POST',
-            headers: {
-                'Authorization': localStorage.jwt,
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(todo)
-        }).then(response => response.json())
-            .then((response) => {
+        saveToDo(todo).then((response) => {
```
[the remark this answers is not in the document]
Comment (kehtolaulu) [reply src/components/toDo/ToDoBox.js:23]: create function where?

File: src/components/toDo/ToDoBox.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (Dimanaux) [reply src/components/toDo/ToDoBox.js:48]: I think we can create
```javascript
const toggleStatus = todo => {
    if (todo.status === ...) { ... } else { ... }
    return updateTodo(todo);
};

File: src/components/toDo/ToDoBox.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (Dimanaux) [reply src/components/toDo/ToDoBox.js:23]: In api.js

File: src/components/toDo/ToDoBox.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (Dimanaux) [reply src/components/toDo/ToDoBox.js:7]: It is a confusing name

File: src/components/toDo/ToDoList.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import ToDo from './ToDo'
+
+class ToDoList extends React.Component {
+    render() {
+        let todoNodes = this.props.data.map(todo => (
```
[the remark this answers is not in the document]
Comment (kehtolaulu) [reply src/components/toDo/ToDoList.js:6]: What does it mean? I don't know what is "to inline variable"

File: src/components/toDo/ToDoForm.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (Dimanaux) [reply src/components/toDo/ToDoForm.js:37]: NO OFFENCE. JUST CODE REVIEW

File: src/components/toDo/ToDoList.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (Dimanaux) [reply src/components/toDo/ToDoList.js:6]: Replace its usage with its declaration.
```javascript
let a = 5;
console.log(a);
// transforms to
console.log(5);
```

File: src/components/auth/SignUp.js (hunk 3 above)
Comment (kehtolaulu) [reply src/components/auth/SignUp.js:25]: ???

**Pull request merged**